store/__tests__/user.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { createLocalVue } from '@vue/test-utils';
+import Vuex, { Store } from 'vuex';
+import { cloneDeep } from 'lodash';
+import { state, mutations, actions, UserState, defaultState } from '../user';
+
+let store: Store<UserState>;
+
+const TEST_ACCESS_TOKEN = 'test-access-token';
+
+beforeEach(() => {
+  createLocalVue().use(Vuex);
+
+  store = new Vuex.Store(cloneDeep({ state, mutations, actions }));
+});
+
+describe('Vuex: user store', () => {
+  it('sets the access token when "SET_USER" is committed', () => {
+    store.replaceState({ ...defaultState() });
+
+    store.commit('SET_USER', { accessToken: TEST_ACCESS_TOKEN });
+
+    expect(store.state.accessToken).toBe(TEST_ACCESS_TOKEN);
+  });
+
+  it('clears the access token When "CLEAR_USER" is committed', () => {
+    store.replaceState({ ...defaultState(), accessToken: TEST_ACCESS_TOKEN });
+
+    store.commit('CLEAR_USER');
+
+    expect(store.state).toMatchObject(defaultState());
+  });
+
+  it('sets the access token when setUser is dispatched', () => {
```

Comment:
This should test for the proper mutation to be called instead.

It can be done by mocking `commit()` and checking if it's called with the proper parameters.

Reply:
We don't mock commit in any of the other store tests. Why change it here?

Reply:
The idea of a unit test is to test things in isolation, so we would test the actions and mutations separately.

This involves being able to see what `commit()` calls, to check if we're calling the proper mutation, with the proper parameters.

Reply:
As for the other tests, some already do. I still need to go over the rest and fix the others, though.